### ISSUE TITLE
Item Durability & Elemental Weakness Mechanics

### ISSUE THREAD

**Issue by Rushwind13** [Rushwind13/JMoria#245]
# Item Durability & Elemental Weakness Mechanics

## Problem
Currently, items are indestructible. Need a **durability system** where items degrade or break when exposed to elemental hazards (fire, cold, acid, lightning). This creates:
- Risk/reward in inventory choices (do I carry Books in a Fire dungeon level?)
- Economic sink (need replacement items, consume consumables faster)
- Realism (books burn, potions boil, metal corrodes)

## Design Philosophy
- **Weakness-based**: Items take damage only from specific elemental attacks
- **Non-permanent**: Can't permanently destroy unique items (prevents soft-locking)
- **Consumable impact**: Scrolls/Potions most vulnerable; Equipment hardier
- **Player choice**: Deliberate item drops for specific level types

### Scrolls (Paper-Based) — Most Vulnerable
| Threat | Effect | Damage |
|---------|---------|---------|
| Fire/Heat | Scroll burns, ink fades | 50-100% durability loss per hit |
| Cold/Freeze | Scroll becomes brittle, tears | 25-50% durability loss |
| Water (future) | Ink bleeds, illegible | 75% durability loss |
| Acid | Paper dissolves | 100% destroyed (unretrievable) |
| Lightning | Charred, crumbly | 50% durability loss |

**Result of Destruction**: Scroll becomes "Illegible Scroll" (unreadable, unsellable, ~0gp)

### Potions (Glass Bottles) — Vulnerable
| Threat | Effect | Damage |
|---------|---------|---------|
| Fire | Bottle boils over, seals pop, potion evaporates | 50-100%, contents lost |
| Cold | Potion freezes or flask cracks | Bottle shatters, 100% loss |
| Acid | Flask corrodes, leaks | 50-75% loss, drips on adjacent items |
| Lightning | Cork flies, contents explode outward | 75% loss, scattered everywhere |
| Crushing (future) | Bottle breaks | 100% loss |

**Result of Destruction**: Potion bottle becomes empty/useless (~0gp), adjacent items take splash damage

### Books (Heavy Paper/Binding) — Moderate Vulnerability
| Threat | Effect | Damage |
|---------|---------|---------|
| Fire | Pages burn, binding chars | 25-50% durability, may become unreadable |
| Cold | Binding warps, pages stick | 15-30% durability |
| Acid | Cover corrodes, pages dissolve | 40-60% durability |
| Water (future) | Pages swell, ink bleeds | 50% durability, readability reduced |
| Lightning | Spine cracks, pages singe | 25% durability |

**Result of Destruction**: Book becomes "Damaged Tome" (unreadable, unusable, ~10% value)

### Equipment (Armor/Weapons) — Hardiest
| Threat | Effect | Damage |
|---------|---------|---------|
| Fire | Metal oxidizes, wooden handles scorch | 5-15% durability, AC reduced by 1-2 |
| Cold | Metal becomes brittle | 5-10% durability, Risk of weapon breaking on critical hits |
| Acid | Metal pits, corrodes joints | 10-20% durability, AC/damage degraded |
| Lightning | Metal arcs and warps | 10-15% durability, enchantments disrupted |
| Crushing (future) | Dents, bends | 5% durability |

**Result of Destruction**: Equipment becomes "Corroded/Damaged [item name]" (functional but penalized AC/damage by 1-3 points)

### Rings & Amulets — Resistant
| Threat | Effect | Damage |
|---------|---------|---------|
| Fire/Cold/Acid | Minimal effect | 1-5% durability loss |
| Lightning | Intrinsic surges, minor damage | 2-5% durability loss |

**Result**: Rings rarely destroyed, mostly superficial (flavor text: "ring glows briefly, unharmed")

### Data Model
```cpp
struct ItemDurability {
  uint16_t current_durability;      // 0-100%
  uint16_t max_durability;          // Usually 100
  ElementType weakness[4];          // Up to 4 elemental weaknesses (FIRE, COLD, ACID, LIGHTNING)
  uint16_t weakness_damage[4];      // % damage per hit from each element (50-100%)
  bool can_be_destroyed;            // Scrolls/Potions: true, Equipment: false
  bool is_destroyed;                // If destroyed, item becomes non-functional
};
```

### Damage Application
```cpp
void ApplyElementalDamage(Item *item, ElementType element, int damage) {
  if (item == nullptr) return;
  
  // Check if item has this weakness
  int weakness_idx = -1;
  for (int i = 0; i < 4; i++) {
    if (item->durability.weakness[i] == element) {
      weakness_idx = i;
      break;
    }
  }
  if (weakness_idx == -1) return; // No weakness, immune
  
  // Calculate damage
  int damage_pct = item->durability.weakness_damage[weakness_idx];
  int durability_loss = (damage * damage_pct) / 100;
  item->durability.current_durability -= durability_loss;
  
  if (item->durability.current_durability <= 0) {
    item->durability.is_destroyed = true;
    item->durability.current_durability = 0;
    // Message: "Your Spell Book has been destroyed by fire!"
  }
  
  if (item->durability.current_durability <= 25 && item->durability.current_durability > 0) {
    DisplayWarning("Your " + item->name + " is nearly destroyed!");
  }
}
```

### Fire-Themed Levels (Red dragons, Demons)
- Red Dragon attacks (Fire breath, Burning touch)
- Fire Elemental spawns
- Floor traps: Lava pits

**Impact**: Scrolls/Potions especially vulnerable; Books reduced 25-50% durability
**Player Adaptation**: Carry fewer scrolls, more direct-action gear (potions for healing instead)

### Cold-Themed Levels (Ice mages, Blue dragons)
- Blue Dragon attacks (Cold breath)
- Ice Elemental spawns
- Floor traps: Frozen ground (slowing)

**Impact**: Potions freeze/shatter, Cold resistance needed
**Player Adaptation**: Use cold resistance intrinsic, careful potion management

### Acid-Themed Levels (Gelatinous creatures, Acid Puddings)
- Ooze attacks (Acid spray)
- Acid Elemental spawns
- Floor traps: Acid pools

**Impact**: Equipment heavily damaged (AC penalties), Items destroyed
**Player Adaptation**: Acid resistance + Astral Plate amulet, or avoid dungeon

### Lightning-Themed Levels (Storm Mages, Djinn)
- Storm Mage casts Lightning Bolt
- Electric Elemental spawns

**Impact**: Equipment slightly damaged, enchantments disrupted temporarily
**Player Adaptation**: Insulation intrinsic, Lightning resistance

### Item Icon Indicators
- **OK**: Normal item sprite (solid color)
- **Warn**: Item sprite with small damage symbol (crack, scorch mark) = <50% durability
- **Critical**: Item sprite with heavy damage symbol = <25% durability
- **Destroyed**: Item sprite greyed out or with "X" overlay = destroyed item

### Inventory Tooltip
```
Spell Book (Damaged)
Durability: [████░░] 45%
Weaknesses: Fire (50% damage), Acid (100% destroyed)
Value: 50 gp (normally 200 gp)
```

### Dungeon Messages
```
> Red Dragon's Fire Breath hits you!
You take 25 damage.
Your Spell Book is damaged! (Durability: 75% → 45%)

> Ooze's Acid Spray touches your Pack!
Your Potion of Cure Wounds has been destroyed!
```

### Unique Items
- Unique items cannot be permanently destroyed (prevents save-scumming)
- But durability still degrades, inflicting penalties (AC, Damage) at <50%
- Unique items restore to 100% durability when character advances 1 level or rests
- **Design rationale**: Hardcore mode needs risk without hard-locks

### Cursed Items
- Cursed items degrade **faster** (weakness damage +50%)
- e.g., "Cursed Spell Book" takes 75% damage from Fire (normally 50%)

### Identify & Pricing
- Destroyed/Damaged items identified as such
- Price reduced: If destroyed = ~5% value, If <50% durability = 40-60% value

## Success Criteria

1. **Risk/Reward**: Elemental damage threatens inventory management
2. **Player Agency**: Deliberate item dropping/carrying choices per level type
3. **Economic Impact**: 10-30% more resource consumption in elemental-heavy levels
4. **No Soft-Locks**: Unique items remain usable (penalized but not broken)
5. **UI Clarity**: Durability status visible at a glance

## Dependencies
- #77 (Effects Framework) — defines elemental damage types
- #128 (Magic Items) — unique item mechanics, damage penalties
- #245 (Monster Color Effects) — elemental attacks identified by monster color
- #242 (Speed System) — affects evade/mitigation of elemental attacks

### Phase 1: Core System
- Data model for durability (current/max, weaknesses, destruction flag)
- Damage application logic (elemental damage → durability loss)
- Basic UI indicators (greyscale overlay for destroyed items)

### Phase 2: Content
- Assign weaknesses to all item types (scrolls/potions/books/equipment)
- Calibrate damage percentages per item type/element
- Add destruction messages and warnings

### Phase 3: Dungeon Balance
- Assign elemental threats to dungeon levels
- Test resource consumption (should increase ~15-20%)
- Polish unique item restoration mechanics

### Phase 4: Polish
- Durability UI icons and tooltips
- Character voice lines (e.g., "My book is burning!")
- Hardcore mode unique item handling (restoration on level-up?)
